server/update-workflow-interface/update-workflow-interface.md

```diff
@@ -0,0 +1,253 @@
+issuer from the server at the time that the Update completes _and_ before the
+issuer becomes aware of the Update ID. The issuer must be able to re-issue the
+same UpdateRequest to access the server-generated Update ID.
+
+## gRPC API
+
+The gRPC API comprises two functions, `UpdateWorkflow` and
+`PollUpdateWorkflowResponse` to be added to the `WorkflowService` service
+definition. The former is used to invoke an update and follows the standard
+patterns for identifying a workflow execution and carrying arbitrary metadata
+(Header) and payloads (Payloads). The request type for this function includes a
+`result_access_style` property indicating whether the caller intends to poll for
+a result later (via `PollUpdateWorkflowResponse`) or wants the result to be
+delivered "inline" as part of the `UpdateWorkflowResponse` that is returned from
+the `UpdateWorkflow` gRPC call. In the case where the client intends to poll,
+the RPC will return as soon as the UpdateID is generated (in such cases, the
+`UpdateWorkflowResponse` may contain the actual UpdateResponse data or may be
+empty with the `UPDATE_WORKFLOW_RESULT_TYPE` flag set to
+`UPDATE_WORKFLOW_RESULT_TYPE_IN_PROGRESS`. If the client wants an inline response value,
+the RPC will block until the response value (or error) is available or the gRPC
+timeout is hit. In either case, the gRPC timeout for this call is server-side
+capped at low value, indicating the intent that this call either complete
+quickly or at least be validated quickly and the client will switch to polling.
+The `request_id` property uniquely identifies the logical request (_not_ the
+issuance of the request) and is intended for use as an idempotency token.
+
+As `UpdateWorkflow` executes workflow code in all cases it is recommended that
+this function be excluded from rpc latency metrics aggregation.
+
+The second function, `PollUpdateWorkflowResponse` is used when a client has
+already issued an `UpdateWorkflow` request and has the resulting `update_id`.
+That `update_id` can be provided to re-attach to an in-flight or completed
+Update and obtain the associated UpdateResult in the form of an
+`UpdateWorkflowResponse`.
+
+
+```protobuf
+service WorkflowService {
+    rpc UpdateWorkflow(UpdateWorkflowRequest) returns (UpdateWorkflowResponse){}
+    rpc PollUpdateWorkflowResponse(PollUpdateWorkflowResponseRequest) returns (UpdateWorkflowResponse){}
+}
+
+message UpdateWorkflowRequest {
+    temporal.api.common.v1.Header header = 1;
+    string request_id = 2;
+    string namespace = 3;
+    temporal.api.common.v1.WorkflowExecution execution = 4;
+    string update_name = 5;
+    temporal.api.common.v1.Payloads input = 6;
+    UpdateWorkflowResultAccessStyle result_access_style = 7;
+}
+
+message UpdateWorkflowResponse {
+    bytes update_id = 1;
+    UpdateWorkflowResultType result_type = 2;
+    UpdateWorkflowSuccess success = 3;
+    UpdateWorkflowFailure failure = 4;
```

Comment:
nit: Consider calling this `failed` and replacing `output` below with `failure`.
We've used this pattern in core and it reads pretty well: `reponse.failed.failure`